src/main/java/com/liveramp/cascading_ext/yarn/YarnApiHelper.java

```diff
@@ -76,14 +79,62 @@ private static String GETRequest(String urlString) throws IOException {
   }
 
   public static Optional<ApplicationInfo> getYarnAppInfo(Configuration conf, String appId) {
-    String yarnApiAddress = conf.get("yarn.resourcemanager.webapp.address", "");
-    if ( !yarnApiAddress.isEmpty()) {
+    String yarnApiAddress = getYarnApiAddress(conf);
+    if (!yarnApiAddress.isEmpty()) {
       return getYarnAppInfo(yarnApiAddress, appId);
     } else {
       return Optional.empty();
     }
   }
 
+  /***
+   * Gets a yarn api address. This also works if you are using more than one resource manager.
+   * If there's more than one resource manager this method will return the address of the first
+   * resource manager with which it was able to successfully connect to. It will return an empty
+   * string if there were no resource manager it could successfully connect to.
+   */
+  private static String getYarnApiAddress(Configuration conf) {
+    Set<String> yarnApiAddresses = getYarnApiAddresses(conf);
+
+    try {
+      for (String yarnApiAddress : yarnApiAddresses) {
+        if (successfulConnection(yarnApiAddress)) {
+          return yarnApiAddress;
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Error getting yarn api address:", e);
+    }
+
+    return "";
+  }
+
+  static Set<String> getYarnApiAddresses(Configuration conf) {
+    String emptyString = "";
+    String rmIdsConf = conf.get("yarn.resourcemanager.ha.rm-ids", emptyString);
+
+    if (rmIdsConf.isEmpty()) {
+      String yarnApiAddress = conf.get("yarn.resourcemanager.webapp.address", emptyString);
+      return yarnApiAddress.isEmpty() ? Sets.newHashSet() : Sets.newHashSet(yarnApiAddress);
+    }
+
+    String[] allRmIds = rmIdsConf.split(",");
+    return Arrays.stream(allRmIds)
+        .map(rmId -> conf.get("yarn.resourcemanager.webapp.address." + rmId, emptyString))
+        .filter(address -> !address.isEmpty())
+        .collect(Collectors.toSet());
+  }
+
+  private static boolean successfulConnection(String urlString) throws IOException {
+    URL url = new URL(urlString);
+    HttpURLConnection urlConnection = (HttpURLConnection)url.openConnection();
+    urlConnection.setConnectTimeout(10000);
```

Comment:
can we make this into one of those fun java versions of `10.seconds`? Assuming this is 10 seconds.

Reply:
`Duration.standardSeconds(10).getMillis()`? Don't you just love the language? :) Could static import the `standardSeconds`, though.

Reply:
yup, exactly, thanks @roshan!